frontend/src/utils/utils.ts

```diff
@@ -30,18 +30,43 @@ export const round = (value: number | string, decimals = 2) => {
 
 /**
  * @notice Rounds to appropriate human readable decimals for the token
- * @dev This is very fragile and should be replace with something robust, i.e. something that walks
- * value and rounds considering sig-figs or similar
+ * @param amount
+ * @param token
+ * @returns string
  */
-export const roundTokenAmount = (amount: BigNumberish, token: { decimals: number; address: string }): string => {
-  const formatted = formatUnits(amount, token.decimals);
-  // WETH on Polygon
-  const isWeth = token.address === '0x7ceB23fD6bC0adD59E62ac25578270cFf1b9f619';
-  if (isWeth) {
-    return round(formatted, 6);
-  } else {
-    return round(formatted, 2);
-  }
+export const humanizeTokenAmount = (
+  amount: BigNumberish,
+  token: TokenInfo,
+): string => {
+  const formattedAmount = parseFloat(formatUnits(amount, token.decimals));
+  if (formattedAmount <= 1) return formattedAmount.toPrecision(2);
+  return formattedAmount.toLocaleString(undefined, {
+    minimumFractionDigits: 0,
+    maximumFractionDigits: 2,
+  })
+};
+
+/**
+ * @notice Rounds the final amount that will be recieved during withdrawal.
+ * For display/UI purposes only -- it doesn't affect the actual amount
+ * withdrawn. Assumes that `amount` and `userFormattedFee` are both denominated in `token`.
+ * @param amount
+ * @param userFormattedFee
+ * @returns string
+ */
+export const roundReceivableAmountAfterFees = (
+  amount: BigNumberish,
+  userFormattedFee: string,
+  token: TokenInfo,
+): string => {
+  const humanizedFeeZeroTrimmed = userFormattedFee.replace(/0+$/, '');
```

Comment:
What does this regex match/do? My check on https://regex101.com/ did not help. Wondering if `parseFloat` or `Number` would work as an alternative?

<img width="210" alt="image" src="https://user-images.githubusercontent.com/17163988/157984589-2a3c2ca4-b5d3-4112-963a-93a531e5c9fc.png">

Reply:
It removes insignificant zeros from the user formatted fee. E.g. if the user formatted fee was `"0.01400"` it would trim this to `"0.014"`. `Number(userFormattedFee).toString()` would do the same thing -- you're right -- but has the downside of Javascript's Number overflow issues.

Reply:
Mind just adding a comment explaining that? Otherwise sgtm